go-controller/pkg/ovn/controller/networkconnect/topology.go

```diff
@@ -0,0 +1,913 @@
+		var err error
+		ops, err = libovsdbops.DeleteLogicalRouterPortWithPredicateOps(c.nbClient, ops, connectRouterName, func(item *nbdb.LogicalRouterPort) bool {
+			return item.ExternalIDs[libovsdbops.ObjectNameKey.String()] == cncName &&
+				item.ExternalIDs[libovsdbops.NetworkIDKey.String()] == strconv.Itoa(networkID)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to delete connect router ports for network %s: %w", owner, err)
+		}
+
+		// 2. Delete network router ports created by this CNC
+		ops, err = libovsdbops.DeleteLogicalRouterPortWithPredicateOps(c.nbClient, ops, networkRouterName, func(item *nbdb.LogicalRouterPort) bool {
+			return item.ExternalIDs[libovsdbops.ObjectNameKey.String()] == cncName &&
+				item.ExternalIDs[libovsdbops.NetworkIDKey.String()] == strconv.Itoa(networkID)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to delete network router ports for network %s: %w", owner, err)
+		}
+
+		// 3. Delete routing policies created by this CNC on the network router
+		ops, err = libovsdbops.DeleteLogicalRouterPolicyWithPredicateOps(c.nbClient, ops, networkRouterName, func(item *nbdb.LogicalRouterPolicy) bool {
+			return item.ExternalIDs[libovsdbops.ObjectNameKey.String()] == cncName &&
+				item.ExternalIDs[libovsdbops.NetworkKey.String()] == owner
+		})
+		if err != nil {
+			return fmt.Errorf("failed to delete routing policies from network router %s: %w", networkRouterName, err)
+		}
+	}
+
+	// Execute all delete operations
+	if len(ops) > 0 {
+		if _, err := libovsdbops.TransactAndCheck(c.nbClient, ops); err != nil {
+			return fmt.Errorf("failed to execute cleanup operations for CNC %s: %w", cncName, err)
+		}
+		klog.Infof("CNC %s: executed %d cleanup operations", cncName, len(ops))
+	}
+
+	return nil
+}
+
+// ensureConnectPortsOps returns ops to create the ports connecting the connect router and network router.
+// For Layer3:
+//   - Local node: creates full port pair (connect-router ↔ network-router) with peer relationship
+//   - Remote nodes: creates only the connect-router side port (with tunnel key, no peer)
+//
+// For Layer2: creates a single port pair (transit router is distributed)
+func (c *Controller) ensureConnectPortsOps(ops []ovsdb.Operation, cnc *networkconnectv1.ClusterNetworkConnect, netInfo util.NetInfo,
+	subnets []*net.IPNet, nodes []*corev1.Node) ([]ovsdb.Operation, error) {
+	cncName := cnc.Name
+	networkName := netInfo.GetNetworkName()
+	connectRouterName := getConnectRouterName(cncName)
+	networkRouterName := netInfo.GetNetworkScopedClusterRouterName()
+	networkID := netInfo.GetNetworkID()
+
+	// Validate subnets are allocated for tunnel key calculation
+	if len(subnets) == 0 {
+		return nil, fmt.Errorf("no subnets allocated for network %s", networkName)
+	}
+
+	var err error
+	if netInfo.TopologyType() == ovntypes.Layer3Topology {
+		// For Layer3 networks, create ports for all nodes
+		for _, node := range nodes {
+			nodeID, err := util.GetNodeID(node)
```

Comment:
so we know that node.Name can never change, and nodeID potentially shouldn't too, but if we imagine nodeID can change,  should we use nodeName instead of nodeID as dbIDs, and then if the nodeID does change, we will jsut update existing objects with the new subnets?

Reply:
ack ill keep this nodeid for now.. when you do a second round in jan we can see if we need to change something.